### PR TITLE
*: better metrics support

### LINKED ISSUE #22: Better metrics support
- Add utility to help start a prometheus push client
- Add grafana metrics template

### PR DESCRIPTION
**Pull request by disksing**
- Add grafana template
- Adjust metrics
- Add `bench` example

Fixes #22

### CODE REVIEW

File: txnkv/txn.go [hunk 1]
```diff
@@ -264,16 +262,19 @@ func (txn *Transaction) Rollback() error {
 	if !txn.valid {
 		return kv.ErrInvalidTxn
 	}
+	start := time.Now()
+	defer func() { metrics.TxnCmdHistogram.WithLabelValues("rollback").Observe(time.Since(start).Seconds()) }()
```
Comment (siddontang) [inline txnkv/txn.go]: em, why not combine them into one defer fun?

File: metrics/metrics.go [hunk 2]
```diff
@@ -34,14 +43,6 @@ var (
 			Help:      "Counter of snapshots.",
 		})
 
-	TxnCmdCounter = prometheus.NewCounterVec(
```
Comment (rleungx) [inline metrics/metrics.go]: Is it not useful anymore?

Comment (disksing) [reply metrics/metrics.go]: Command duration already contains count so it can be removed.

File: txnkv/txn.go (hunk 1 above)
Comment (disksing) [reply txnkv/txn.go]: good idea

**Pull request merged**